--- gcc-13.2.0/gcc/config/mips/t-mti-elf	2023-07-27 11:13:04.000000000 +0300
+++ gcc-13.2.0/gcc/config/mips/t-mti-elf-new	2024-04-06 05:57:41.083277145 +0300
@@ -19,8 +19,8 @@
 # The default build is mips32r2, hard-float big-endian.  Add mips32,
 # soft-float, and little-endian variations.
 
-MULTILIB_OPTIONS = mips32/mips64/mips64r2 mips16/mmicromips mabi=64 EL msoft-float mnan=2008
-MULTILIB_DIRNAMES = mips32 mips64 mips64r2 mips16 micromips 64 el sof nan2008
+MULTILIB_OPTIONS = mips32/mips32r6/mips64/mips64r2/mips64r6 mabi=64 EL msoft-float
+MULTILIB_DIRNAMES = mips32 mips32r6 mips64 mips64r2 mips64r6 64 el sof
 MULTILIB_MATCHES = EL=mel EB=meb mips32r2=mips32r3 mips32r2=mips32r5 mips64r2=mips64r3 mips64r2=mips64r5
 
 # The 64 bit ABI is not supported on the mips32 architecture.
@@ -32,17 +32,4 @@
 # architecture flag instead.
 MULTILIB_EXCEPTIONS += mabi=64*
 
-# We do not want to build mips16 versions of mips64* architectures.
-MULTILIB_EXCEPTIONS += *mips64*/*mips16*
-MULTILIB_EXCEPTIONS += *mips16/mabi=64*
-
-# We only want micromips for mips32r2 architecture.
-MULTILIB_EXCEPTIONS += *mips32/mmicromips*
-MULTILIB_EXCEPTIONS += *mips64*/mmicromips*
-MULTILIB_EXCEPTIONS += *mmicromips/mabi=64*
-
-# We do not want nan2008 libraries for soft-float,
-# mips32[r1], or mips64[r1].
-MULTILIB_EXCEPTIONS += *msoft-float*/*mnan=2008*
-MULTILIB_EXCEPTIONS += *mips32/*mnan=2008*
-MULTILIB_EXCEPTIONS += *mips64/*mnan=2008*
+MULTILIB_EXCLUSIONS += mips64/!mabi=64 mips64r2/!mabi=64 mips64r6/!mabi=64
